Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -114,6 +115,20 @@ public void SerializesAndDeserializesSubtypesBasedOnTypeAnnotations()
             );
         }
 
+        [TestMethod]
+        public void DeserializeMaxDepth()
+        {
+            var stream = new IonSerializer(new IonSerializationOptions {IncludeFields = true}).Serialize(TestObjects.drKyler);
+            
+            var serializer = new IonSerializer(new IonSerializationOptions {MaxDepth = 2, IncludeFields = true});
+            var deserialized = serializer.Deserialize<Teacher>(stream);
+
+            Assert.IsNotNull(deserialized.car);
```

Comment:
This becomes a one-liner and more expressive if you use the string comparison method I've used for all the other tests.

Reply:
`Check()` and `Serde()` won't work here because they use the same serializer for serialization and deserialization. 
For this test we want to fully serialize something and then deserialize it at a lower MaxDepth so there will be some fields left un-deserialized.

Reply:
So you can just new-up your own instance with the null fields.

Reply:
But that wouldn't really test the MaxDepth option.

If the original input already has null fields, then we would expect the deserialized output to also have those null fields regardless of the MaxDepth option.

Or did I misunderstand your suggestion?